modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/bookmark/BookmarksPage.kt

```diff
@@ -155,26 +164,48 @@ private fun BookmarksView(
     onRowLongPressed: (Bookmark) -> Unit,
     onOptionsMenuClicked: () -> Unit,
     onPlayClick: (Bookmark) -> Unit,
+    onSearchTextChanged: (String) -> Unit,
 ) {
+    val focusRequester = remember { FocusRequester() }
     LazyColumn(
         modifier = Modifier
             .fillMaxSize(),
     ) {
-        item {
-            val title = stringResource(
-                id = if (state.bookmarks.size > 1) {
-                    LR.string.bookmarks_plural
-                } else {
-                    LR.string.bookmarks_singular
-                },
-                state.bookmarks.size,
-            )
+        if (state.searchEnabled) {
+            item {
+                SearchBar(
+                    text = state.searchText,
+                    placeholder = stringResource(LR.string.search),
+                    onTextChanged = onSearchTextChanged,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(horizontal = 16.dp, vertical = 8.dp)
+                        .focusRequester(focusRequester),
+                )
+            }
+        }
+        if (state.searchEnabled &&
+            state.searchText.isNotEmpty() &&
+            state.bookmarks.isEmpty()
+        ) {
+            item { NoBookmarksInSearchView(onActionClick = { onSearchTextChanged("") }) }
+        } else {
+            item {
+                val title = stringResource(
+                    id = if (state.bookmarks.size > 1) {
+                        LR.string.bookmarks_plural
+                    } else {
+                        LR.string.bookmarks_singular
```

Comment:
I realize this is already in our app but shouldn't this be handles with [plural strings](https://developer.android.com/guide/topics/resources/string-resource#Plurals) and not regular strings? Otherwise it doesn't translate well to different languages that have different grammar rules for different quantities.

Reply:
I don't know the full context but it could be due to GlotPress not supporting plurals yet. I'll leave it for now and we can discuss it internally.